sdk/python/kfp/compiler/helpers.py

```diff
@@ -0,0 +1,448 @@
+# Copyright 2021-2022 The Kubeflow Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Helper functions for DSL compiler."""
```

Comment:
What do you think about merging this into `pipeline_spec_builder`, so that we could have a central place for or IR related manipulation?

Reply:
Good idea. Thanks, Chen. This is done.